udata/core/discussions/caches.py

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from flask import url_for
+
+from udata.app import cache
+
+from .signals import (
+    on_new_discussion, on_new_discussion_comment, on_discussion_closed,
+    on_discussion_deleted
+)
+
+
+@on_new_discussion.connect
+@on_new_discussion_comment.connect
+@on_discussion_closed.connect
+@on_discussion_deleted.connect
+def invalidate_discussions_cache(discussion, **kwargs):
+    cache_key = 'discussions{url}?for={id}'.format(
```

Comment:
You have to take into account the `key_pattern` passed but I'll double check that.

I think we can find a better way to generate cache keys but I'm still hesitating on the right place to put it and how to deal with GET params everywhere.